lib/puppet.rb

```diff
@@ -258,6 +258,11 @@ def self.rollback_context(name)
 
   # The single instance used for normal operation
   @context = Puppet::Context.new(bootstrap_context)
+
+  def self.future_parser?
+    env = Puppet.lookup(:current_environment) { return Puppet[:parser] == 'future' }
+    Puppet.lookup(:environments).get_conf(env.name).parser == 'future'
```

Comment:
get_conf returns nil if it is not a directory env. In that case, return Puppet[:parser]